datadog/resource_datadog_monitor.go

```diff
@@ -268,9 +268,32 @@ func resourceDatadogMonitor() *schema.Resource {
 				Default:     true,
 			},
 			"locked": {
-				Description: "A boolean indicating whether changes to to this monitor should be restricted to the creator or admins. Defaults to `false`.",
-				Type:        schema.TypeBool,
-				Optional:    true,
+				Description:   "A boolean indicating whether changes to to this monitor should be restricted to the creator or admins. Defaults to `false`.",
+				Type:          schema.TypeBool,
+				Optional:      true,
+				ConflictsWith: []string{"restricted_roles"},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
Why do we need those diff suppress if it can't be set by terraform? Is it returned by the API somehow?

Reply:
Ya, there's some logic that if restricted roles are set, we also return `locked` as true (regardless of its value in the request). This is in theory supposed to help with migration across tools.